src/linear_solver_routines/block_solver_helpers/gentlemans_house.jl

```diff
@@ -0,0 +1,161 @@
+# This is file from RLinearAlgebra.jl
+"""
+    GentData{S, M<:Matrix{S}, V<:Vector{S}}
+
+This is a preallocated data structure for solving least squares problems with 
+Gentleman's algorithm. This data structure is not exported and thus is not 
+designed to be manipulated by the user.
+# Fields
+- `A::M`, The matrix to be solved.
+- `B::M`, The matrix that rows of `A` are brought to have QR decomposition
+applied.
```

Comment:
This is unclear.

Reply:
I changed this to be a block of rows of A that are used to update the qr decomposition